apps/server/src/modules/learnroom/uc/dashboard.uc.ts

```diff
@@ -31,7 +54,8 @@ export class DashboardUc {
 		const dashboard = await this.dashboardRepo.getDashboardById(dashboardId);
 		this.validateUsersMatch(dashboard, userId);
 
-		dashboard.moveElement(from, to);
+		const [substituteCourses] = await this.courseRepo.findAllForSubstituteTeacher(userId); // in the future this will be removed because it is used as a temporory solution just to stop arrrnagement and grouping of substitute courses
+		dashboard.moveElement(from, to, substituteCourses);
```

Comment:
I think it is a lot cleaner to simply move substitudeCousers in the moveElement method, instead of getting it here and then passing it as argument, or what am I not seeing here
as for the comment, best case you should leave a TODO. We have plenty of technical depth with the legacy, any solution we implement new must assume it can stay like this for a long time

Reply:
Ok so moveElement method is in dashboard entity class. I don't think that we want to inject any repos (courseRepo) to entity classes that's the main reason why it is in that place then passed as an argument to move element method